src/main/java/com/example/elasticagent/requests/MigrateConfigPayload.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2017 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.elasticagent.requests;
+
+import com.example.elasticagent.*;
+import com.google.gson.FieldNamingPolicy;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.annotations.Expose;
+
+import java.util.List;
+import java.util.Map;
+
+import static com.example.elasticagent.ExamplePlugin.GSON;
+
+public class MigrateConfigPayload {
+
+
+    @Expose
+    private PluginSettings pluginSettings;
+
+    @Expose
+    private List<Map<String, String>> elasticAgentProfiles;
+
+    @Expose
+    private List<ClusterProfile> clusterProfiles;
```

Comment:
This won't work! as for all the other requests, we send only cluster Profile properties. For migrate config request, we sent id, plugin-id, and properties.

Refer to [this](https://plugin-api.gocd.org/19.3.0/elastic-agents/#migrate-config) documentation for more information.

Reply:
Done.